Modules/BarcodeScanner/BarcodeScannerView.swift

```diff
@@ -0,0 +1,320 @@
+        }
+    }
+
+    @objc
+    private func statusBarOrientationDidChange(_ notification: NSNotification) {
+        guard let rawInterfaceOrientation = notification.userInfo?[UIApplication.statusBarOrientationUserInfoKey] as? Int else { return }
+        guard let interfaceOrientation = UIInterfaceOrientation(rawValue: rawInterfaceOrientation) else { return }
+
+        setVideoOrientation(to: interfaceOrientation)
+    }
+
+    @objc
+    private func captureDeviceSubjectAreaDidChange(_ notification: NSNotification) {
+        captureSessionQueue.async { [weak self] in
+            guard let self = self, let device = self.captureDeviceInput?.device else { return }
+
+            do {
+                try device.lockForConfiguration()
+                defer { device.unlockForConfiguration() }
+
+                if device.isFocusPointOfInterestSupported, device.isFocusModeSupported(.continuousAutoFocus) {
+                    device.focusPointOfInterest = CGPoint(x: 0.5, y: 0.5)
+                    device.focusMode = .continuousAutoFocus
+                }
+
+                if device.isExposurePointOfInterestSupported && device.isExposureModeSupported(.continuousAutoExposure) {
+                    device.exposurePointOfInterest = CGPoint(x: 0.5, y: 0.5)
+                    device.exposureMode = .continuousAutoExposure
+                }
+
+                device.isSubjectAreaChangeMonitoringEnabled = false
+            } catch {
+                DispatchQueue.main.async {
+                    self.model.onError(.deviceLockingForConfigurationFailed)
+                }
+            }
+        }
+    }
+
+    @objc
+    private func captureSessionRuntimeError(_ notification: NSNotification) {
+        guard let errorValue = notification.userInfo?[AVCaptureSessionErrorKey] as? NSError else { return }
+
+        let error = AVError(_nsError: errorValue)
+        if error.code == .mediaServicesWereReset {
+            captureSessionQueue.async { [weak self] in
+                guard let self = self, self.isCaptureSessionRunning else { return }
+
+                self.captureSession.startRunning()
+                self.isCaptureSessionRunning = self.captureSession.isRunning
+            }
+        } else {
+            model.onError(.runtimeError(error))
+        }
+    }
+}
+
+extension BarcodeScannerView: AVCaptureMetadataOutputObjectsDelegate {
+    public func metadataOutput(
+        _ output: AVCaptureMetadataOutput,
+        didOutput metadataObjects: [AVMetadataObject],
+        from connection: AVCaptureConnection
+    ) {
+        let barcodes = metadataObjects.compactMap { metadataObject -> Barcode? in
+            guard let machineReadableCodeObject = metadataObject as? AVMetadataMachineReadableCodeObject else { return nil }
+            guard let type = BarcodeType(rawValue: machineReadableCodeObject.type) else { return nil }
+
+            return Barcode(type: type, body: machineReadableCodeObject.stringValue ?? "")
```

Comment:
Should we show barcode scanning results if the found `machineReadableCodeObject.stringValue` is `nil`? Or should we just ignore them and don't return an empty string here?

Reply:
I don't think that it is possible in general but we don't have to consider that situation here because we'll never return an empty one then.